content/doc/tutorials/using-jenkinsfile-runner-github-action-to-build-jenkins-pipeline.adoc

```diff
@@ -0,0 +1,364 @@
+NOTE: This option and its core are still in progress, so it’s not mentioned in the Jenkinsfile Runner GitHub Actions official guide.
+However, it does work and can be used at this time.
+
+In this example, we review how to use Groovy scripts to set up the Jenkins instance:
+
+. Create a directory, for example “groovy.init.d”, to store all your Groovy setup scripts.
+. Create a Groovy file called “test.groovy”.
+* Do not name it `init.groovy` because this name is already occupied.
+. Then, add the debug output:
++
+[source,groovy]
+----
+println 'Hello Groovy Hooks!'
+----
+. Specify the “groovy.init.d” directory in the GitHub Actions input.
++
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+  initHook: groovy.init.d
+----
+. Check the GitHub Actions log and verify that the groovy script is executed right after Jenkins starts up, before the actual pipeline is run.
+
+=== Integrate with other GitHub Actions (Optional)
+
+You can integrate this process with other GitHub Actions in the marketplace, via Jenkinsfile Runner GitHub Actions.
+However, if the starting time of the Jenkins container is different in these GitHub Actions, some GitHub Actions cannot be used. 
+In other words, `jfr-static-image-action` cannot be integrated with the other GitHub Actions except `actions/checkout`, but you can integrate other GitHub Actions with `jfr-container-action` and `jfr-runtime-action`. 
+Refer to their differences in link:https://jenkinsci.github.io/jfr-action-doc/docs/user-guide/actions-cmp.html[the official guide].
+
+In this example, we show how to integrate actions/setup-node GitHub Action with jfr-runtime-action to compile a JavaScript project:
+
+. Use an Ubuntu runner for the job.
++
+[source,yaml]
+----
+jobs:
+  job-name:
+    runs-on: ubuntu-latest
+----
+. Call the `actions/checkout@v2` to pull your codes into the runner.
++
+[source,yaml]
+----
+- uses: actions/checkout@v2
+----
+. Call the `actions/setup-node@v3` to set up node 18.
++
+[source,yaml]
+----
+- uses: actions/setup-node@v3
+  with:
+    node-version: 18
+----
+. Call the `jenkinsci/jfr-setup-action@master` to set up Jenkins.
++
+[source,yaml]
+----
+- uses: jenkinsci/jfr-setup-action@master
+----
+. Call the `jenkinsci/jfr-plugin-installation-action@master` to install additional plugins.
++
+[source,yaml]
+----
+- uses: jenkinsci/jfr-plugin-installation-action@master
+  with:
+      pluginstxt: plugins.txt
+----
+. Call the `jenkinsci/jfr-runtime-action@master` to run the Jenkins pipeline.
++
+[source,yaml]
+----
+- uses: jenkinsci/jfr-runtime-action@master
+  with:
+    command: run
+    jenkinsfile: Jenkinsfile
+----
+
+Refer to the link:https://github.com/jenkinsci/jfr-action-demo/tree/master/demo/javascript/my-react-app[official repository] for the full example.
```

Comment:
```suggestion
Refer to the link:https://github.com/jenkinsci/jfr-action-demo/tree/master/demo/javascript/my-react-app[official repository] for the full example.
```
If this ends up being a blog post, it would be ideal to have a small conclusion/wrap up after this final section. This documentation hard stops after the final section, but this is more dependent on the format of the post than anything else.

Reply:
Thanks for your advice! It does end abruptly as a tutorial. Typically, I only want the developers to follow the necessary parts (the sections without **optional**) to achieve their goals. When it comes to other further task, they can follow the optional sections and official guide. I'll add the **Wrapping up** section later.

Reply:
Resloved this problem in the latest commits.